packages/mdc-textfield/index.ts

```diff
@@ -153,13 +121,20 @@ class MDCTextField extends MDCComponent {
 
     this.ripple = null;
     if (!this.root_.classList.contains(cssClasses.TEXTAREA) && !this.root_.classList.contains(cssClasses.OUTLINED)) {
-      const MATCHES = getMatchesProperty(HTMLElement.prototype);
-      const adapter =
-        Object.assign(MDCRipple.createAdapter(/** @type {!RippleCapableSurface} */ (this)), {
-          isSurfaceActive: () => this.input_[MATCHES](':active'),
-          registerInteractionHandler: (type, handler) => this.input_.addEventListener(type, handler),
-          deregisterInteractionHandler: (type, handler) => this.input_.removeEventListener(type, handler),
-        });
+      const adapter = {
+        ...MDCRipple.createAdapter(this),
+        ...({
+          // tslint:disable:object-literal-sort-keys
+          isSurfaceActive: () => ponyfill.matches(this.input_, ':active'),
+          registerInteractionHandler: (type: string, handler: EventListener) => {
```

Comment:
These ones can be removed, but the ones in `getInputAdapterMethods_()` need to remain because that method returns an anonymous object. I tried typing the methods as `Partial<MDCTextFieldAdapter>` but that blew up `tsc`.